src/code_1b40_len_20b0.c

```diff
@@ -34,9 +35,155 @@ Gfx D_80074230[] = {
+            *temp80074021 = D_80074020;
+        } else {
+            return;
+        }
+    }
+
+    func_8011BAE8();
+    func_8003857C();
+    update_dynamic_entities();
+    update_triggers();
+    update_scripts();
+    update_messages();
+    update_menu_icons();
+    step_current_game_mode();
+    update_entities();
+    func_80138198();
+    func_8014A548();
+    update_ambient_sounds();
+    func_80149734();
+    update_windows();
+    update_curtains();
+
+    overrideFlags = &gOverrideFlags;
+
+    if (*overrideFlags & 0x20) {
+        temp80074182 = &D_800741A2;
+
+        switch (*temp80074182) {
+            case 0:
+                *overrideFlags |= 0x200;
+                disable_player_input();
+                temp800741A0 = &D_800741A0;
+                if (*temp800741A0 == 255) {
+                    *temp80074182 = 1;
+                    D_8009A690 = 3;
+                } else {
+                    *temp800741A0 += 10;
+                    if (*temp800741A0 > 255) {
+                        *temp800741A0 = 255;
+                    }
+                }
+                break;
+            case 1:
+                *overrideFlags |= 0x8;
+                temp8009A690 = &D_8009A690;
+                (*temp8009A690)--;
+                if (*temp8009A690 == 0) {
+                    func_80149838();
+                    set_game_mode(0);
+                    *overrideFlags &= ~0x20;
+                }
+                break;
+        }
+    } else {
+        D_800741A0 = 0;
+        D_800741A2 = 0;
+    }
+
+    {
+        s32* overrideFlags = &gOverrideFlags;
+
+        if (*overrideFlags & 0x100) {
+            *overrideFlags |= 0x1000;
+        } else {
+            *overrideFlags &= ~0x1000;
+        }
+    }
+
+    {
+        s32* overrideFlags = &gOverrideFlags;
+
+        if (*overrideFlags & 0x200) {
+            *overrideFlags |= 0x2000;
+        } else {
+            *overrideFlags &= ~0x2000;
+        }
+    }
+
+    {
+        s32* overrideFlags = &gOverrideFlags;
+
+        if (*overrideFlags & 0x400) {
+            *overrideFlags |= 0x4000;
+        } else {
+            *overrideFlags &= ~0x4000;
+        }
+    }
+
+    {
+        s32* overrideFlags = &gOverrideFlags;
+
+        if (*overrideFlags & 0x800) {
+            *overrideFlags |= 0x8000;
+        } else {
+            *overrideFlags &= ~0x8000;
+        }
+    }
+
+    rand_int(1);
+}
+
+void gfx_task_background(void) {
+    DisplayContext** gDisplayContextPtr = &gDisplayContext;
+    Gfx** gfx = &gMasterGfxPos;
+    DisplayContext* dispContexts = D_80164000;
+
+    *gDisplayContextPtr = &dispContexts[gCurrentDisplayContextIndex];
+    *gfx = &(*gDisplayContextPtr)->backgroundGfx[0];
+
+    gfx_init_state();
+    gfx_draw_background();
+
+    gDPFullSync((*gfx)++);
+    gSPEndDisplayList((*gfx)++);
+
+    ASSERT((s32)((u32)((*gfx) - (*gDisplayContextPtr)->backgroundGfx) << 3 >> 3) < ARRAY_COUNT((*gDisplayContextPtr)->backgroundGfx));
```

Comment:
what

Reply:
https://discord.com/channels/279322074412089344/704241951972524063/822051304314241034

gonna leave a note but yeah this is a workaround to something that is not quite right